src/Datadog.Trace.ClrProfiler.Managed/Conventions/Convention.cs

```diff
@@ -0,0 +1,31 @@
+using Datadog.Trace.Configuration;
+
+namespace Datadog.Trace.ClrProfiler.Conventions
+{
+    internal struct Convention
+    {
+        public IOutboundHttpConvention OutboundHttpScopeConvention;
+
+        private static Convention singleton;
+        private static volatile bool initialized;
+        private static object syncRoot = new object();
+
+        public static Convention Get(TracerSettings settings)
+        {
+            if (!initialized)
```

Comment:
We could use Lazy for singleton, the sync object would not be needed.

Reply:
No longer valid: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/63/commits/c81a57ae52909d614a7bd7b58b83efb99ec2385d